apps/cms.c

```diff
@@ -885,10 +892,30 @@ int cms_main(int argc, char **argv)
             goto end;
     }
 
-    in = bio_open_default(infile, 'r',
-                          binary_files ? FORMAT_BINARY : informat);
-    if (in == NULL)
-        goto end;
+    if (digesthex != NULL) {
+        if (infile != NULL || !(flags & CMS_DETACHED)) {
+            BIO_printf(bio_err,
+                       "Cannot use -digest when -in or -nodetach is given\n");
+            goto end;
+        }
+        if (operation != SMIME_SIGN) {
+            BIO_printf(bio_err,
+                       "Warning: -digest is ignored for non-signing "
+                       "operation\n");
```

Comment:
The coding guidelines say in https://github.com/openssl/technical-policies/blob/master/policies/coding-style.md#chapter-2-breaking-long-lines-and-strings
> Never break user-visible strings, however, because that breaks the ability to grep for them.

which takes precedence over avoiding long source code lines.